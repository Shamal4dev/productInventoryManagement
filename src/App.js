import React from 'react';
import { Route, Switch, Redirect, withRouter } from 'react-router-dom';
import { Header, HomePage, Footer } from './view/sharedComponents';
import { PrivateRoute } from './view/PrivateRoute';
import { LoginPage, RegisterPage } from './view/usersComponents';
import { connect } from 'react-redux';
import { alertActions } from './script/redux';

import { Test } from './view/LoginPage copy';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.props.history.listen((location, action) => {
      // clear alert on location change
      this.props.clearAlerts();
    });
  }
  render() {
    const { alert } = this.props;
    return (
      <div className='h-100'>
        <Header />
        <div className="jumbotron mb-0">
          <div className="container">
            <div >
              {alert.message &&
                <div className={`alert ${alert.type}`}>{alert.message}</div>
              }

              <Switch>
                <PrivateRoute exact path="/about" component={HomePage} />
                <PrivateRoute exact path="/desc" component={HomePage} />
              
                <Route exact path="/" component={HomePage}></Route>
                <Route path="/login" component={LoginPage}></Route>
                <Route path="/register" component={RegisterPage}></Route>
                <Route path="/test" component={Test}></Route>
                <Redirect from="*" to="/login" />
              </Switch>

            </div>
          </div>
        </div>

        <Footer />
      </div>
    )
  }
}

function mapState(state) {
  const { alert } = state;
  return { alert };
}

const actionCreators=  {
  clearAlerts: alertActions.clear
}

const connectedApp = withRouter(connect(mapState, actionCreators)(App));
export { connectedApp as App };
