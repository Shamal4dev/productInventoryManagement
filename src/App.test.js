import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';

const props = {
    store: {

    },
    history: {
        listen: () => { }
    },
    alert: {
        message: ''
    },
    getAllproducts: ()=> {}
};
describe('App snapshot', () => {

    let wrapper = shallow(
        <App {...props} />
    );

    test('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});