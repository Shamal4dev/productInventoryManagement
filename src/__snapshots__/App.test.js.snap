// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders correctly 1`] = `
<div
  className="h-100"
>
  <Header />
  <div
    className="jumbotron mb-0"
  >
    <div
      className="container"
    >
      <div>
        <Switch>
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Product)",
                "type": [Function],
              }
            }
            exact={true}
            path="/product/:id"
          />
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(DeleteProduct)",
                "type": [Function],
              }
            }
            exact={true}
            path="/deleteProduct"
          />
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(DeleteProduct)",
                "type": [Function],
              }
            }
            exact={true}
            path="/deleteProduct/:id"
          />
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(AddPage)",
                "type": [Function],
              }
            }
            exact={true}
            path="/addProduct"
          />
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(EditPage)",
                "type": [Function],
              }
            }
            exact={true}
            path="/updateProduct"
          />
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(EditPage)",
                "type": [Function],
              }
            }
            exact={true}
            path="/updateProduct/:id"
          />
          <PrivateRoute
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(UserDetail)",
                "type": [Function],
              }
            }
            exact={true}
            path="/userDetails"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Home)",
                "type": [Function],
              }
            }
            exact={true}
            path="/"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(LoginPage)",
                "type": [Function],
              }
            }
            exact={true}
            path="/login"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Register)",
                "type": [Function],
              }
            }
            exact={true}
            path="/register"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />
          <Redirect
            from="*"
            to="/login"
          />
        </Switch>
        <br />
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
